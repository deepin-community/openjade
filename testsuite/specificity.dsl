<!DOCTYPE style-sheet PUBLIC "-//James Clark//DTD DSSSL Style Sheet//EN">

<style-specification id=test1 use=test2>

(define debug (external-procedure "UNREGISTERED::James Clark//Procedure::debug"))

(root 
  (make simple-page-sequence (process-children)))

(element test (with-mode test-mode (process-children)))


;; test specificity rules

; this was necessary to avoid bugs in subgrove
;(define (doc-root) (node-property 'docelem (current-root))) 
(define doc-root current-root)

(mode test-mode

(query 
  (q-element 'A (doc-root))
  (make paragraph (literal "query (priority 1) matching " (gi (current-node))))
  1)
  
(query 
  (node-list 
    (q-element 'A (doc-root)) 
    (q-element 'B (doc-root)))
  (make paragraph (literal "query (priority 0) matching " (gi (current-node)))))
  
(query 
  (node-list 
    (q-element 'A (doc-root)) 
    (q-element 'B (doc-root)) 
    (q-element 'C (doc-root))) 
  (make paragraph (literal "query (priority -1) matching " (gi (current-node))))
  -1)
  
(id A (make paragraph (literal "id matching " (gi (current-node)))))
(id B (make paragraph (literal "id matching " (gi (current-node)))))
(id C (make paragraph (literal "id matching " (gi (current-node)))))
(id D (make paragraph (literal "id matching " (gi (current-node)))))

(element A (make paragraph (literal "element matching " (gi (current-node)))))
(element B (make paragraph (literal "element matching " (gi (current-node)))))
(element C (make paragraph (literal "element matching " (gi (current-node)))))
(element D (make paragraph (literal "element matching " (gi (current-node)))))
(element E (make paragraph (literal "element matching " (gi (current-node)))))

) ; end of test mode

(query 
  (node-list 
    (q-element 'A (doc-root)) 
    (q-element 'B (doc-root)) 
    (q-element 'C (doc-root)) 
    (q-element 'D (doc-root)) 
    (q-element 'E (doc-root)) 
    (q-element 'F (doc-root))) 
  (make paragraph (literal "initial mode query (priority 1) matching " (gi (current-node))))
  1)
  

(query 
  (node-list 
    (q-element 'A (doc-root)) 
    (q-element 'B (doc-root)) 
    (q-element 'C (doc-root)) 
    (q-element 'D (doc-root)) 
    (q-element 'E (doc-root)) 
    (q-element 'F (doc-root)) 
    (q-element 'G (doc-root))) 
  (make paragraph (literal "initial mode query (priority 0) matching " (gi (current-node)))))
  
(query 
  (node-list 
    (q-element 'A (doc-root)) 
    (q-element 'B (doc-root)) 
    (q-element 'C (doc-root)) 
    (q-element 'D (doc-root)) 
    (q-element 'E (doc-root)) 
    (q-element 'F (doc-root)) 
    (q-element 'G (doc-root)) 
    (q-element 'H (doc-root))) 
  (make paragraph (literal "initial mode query (priority -1) matching " (gi (current-node))))
  -1)
  
(id A (make paragraph (literal "initial mode id matching " (gi (current-node)))))
(id B (make paragraph (literal "initial mode id matching " (gi (current-node)))))
(id C (make paragraph (literal "initial mode id matching " (gi (current-node)))))
(id D (make paragraph (literal "initial mode id matching " (gi (current-node)))))
(id E (make paragraph (literal "initial mode id matching " (gi (current-node)))))
(id F (make paragraph (literal "initial mode id matching " (gi (current-node)))))
(id G (make paragraph (literal "initial mode id matching " (gi (current-node)))))
(id H (make paragraph (literal "initial mode id matching " (gi (current-node)))))
(id I (make paragraph (literal "initial mode id matching " (gi (current-node)))))

(element A (make paragraph (literal "initial mode element matching " (gi (current-node)))))
(element B (make paragraph (literal "initial mode element matching " (gi (current-node)))))
(element C (make paragraph (literal "initial mode element matching " (gi (current-node)))))
(element D (make paragraph (literal "initial mode element matching " (gi (current-node)))))
(element E (make paragraph (literal "initial mode element matching " (gi (current-node)))))
(element F (make paragraph (literal "initial mode element matching " (gi (current-node)))))
(element G (make paragraph (literal "initial mode element matching " (gi (current-node)))))
(element H (make paragraph (literal "initial mode element matching " (gi (current-node)))))
(element I (make paragraph (literal "initial mode element matching " (gi (current-node)))))
(element J (make paragraph (literal "initial mode element matching " (gi (current-node)))))

<style-specification id=test2>

(query 
  (node-list 
    (q-element 'A (doc-root)) 
    (q-element 'B (doc-root)) 
    (q-element 'C (doc-root)) 
    (q-element 'D (doc-root)) 
    (q-element 'E (doc-root)) 
    (q-element 'F (doc-root)) 
    (q-element 'G (doc-root)) 
    (q-element 'H (doc-root)) 
    (q-element 'I (doc-root)) 
    (q-element 'J (doc-root)) 
    (q-element 'K (doc-root))) 
  (make paragraph (literal "part 2 initial mode query (priority 1) matching " (gi (current-node))))
  1)
  

(query 
  (node-list 
    (q-element 'A (doc-root)) 
    (q-element 'B (doc-root)) 
    (q-element 'C (doc-root)) 
    (q-element 'D (doc-root)) 
    (q-element 'E (doc-root)) 
    (q-element 'F (doc-root)) 
    (q-element 'G (doc-root)) 
    (q-element 'H (doc-root)) 
    (q-element 'I (doc-root)) 
    (q-element 'J (doc-root)) 
    (q-element 'K (doc-root)) 
    (q-element 'L (doc-root))) 
  (make paragraph (literal "part 2 initial mode query (priority 0) matching " (gi (current-node)))))
  
(query 
  (node-list 
    (q-element 'A (doc-root)) 
    (q-element 'B (doc-root)) 
    (q-element 'C (doc-root)) 
    (q-element 'D (doc-root)) 
    (q-element 'E (doc-root)) 
    (q-element 'F (doc-root)) 
    (q-element 'G (doc-root)) 
    (q-element 'H (doc-root)) 
    (q-element 'I (doc-root)) 
    (q-element 'J (doc-root)) 
    (q-element 'K (doc-root)) 
    (q-element 'L (doc-root)) 
    (q-element 'M (doc-root))) 
  (make paragraph (literal "part 2 initial mode query (priority -1) matching " (gi (current-node))))
  -1)
  
(id A (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id B (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id C (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id D (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id E (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id F (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id G (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id H (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id I (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id J (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id K (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id L (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id M (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))
(id N (make paragraph (literal "part 2 initial mode id matching " (gi (current-node)))))

(element A (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element B (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element C (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element D (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element E (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element F (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element G (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element H (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element I (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element J (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element K (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element L (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element M (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element N (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))
(element O (make paragraph (literal "part 2 initial mode element matching " (gi (current-node)))))

(default (make paragraph (literal "part 2 initial mode default matching " (gi (current-node)))))

