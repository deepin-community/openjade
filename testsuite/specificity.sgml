<!doctype test [
<!entity % elts "(a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p)" >
<!element test - - %elts; >
<!element %elts; - o empty > 
<!attlist %elts id ID #required > 
]>
<test>
<a id=a><b id=b><c id=c><d id=d><e id=e><f id=f><g id=g><h id=h>
<i id=i><j id=j><k id=k><l id=l><m id=m><n id=n><o id=o><p id=p>
</test>
