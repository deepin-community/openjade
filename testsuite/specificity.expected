<?xml version="1.0"?>
<fot>
<simple-page-sequence>
<paragraph>
<a name="0"/>
<text>query (priority 1) matching A</text>
</paragraph>
<paragraph>
<text>query (priority 0) matching B</text>
</paragraph>
<paragraph>
<text>query (priority -1) matching C</text>
</paragraph>
<paragraph>
<text>id matching D</text>
</paragraph>
<paragraph>
<text>element matching E</text>
</paragraph>
<paragraph>
<text>initial mode query (priority 1) matching F</text>
</paragraph>
<paragraph>
<text>initial mode query (priority 0) matching G</text>
</paragraph>
<paragraph>
<text>initial mode query (priority -1) matching H</text>
</paragraph>
<paragraph>
<text>initial mode id matching I</text>
</paragraph>
<paragraph>
<text>initial mode element matching J</text>
</paragraph>
<paragraph>
<text>part 2 initial mode query (priority 1) matching K</text>
</paragraph>
<paragraph>
<text>part 2 initial mode query (priority 0) matching L</text>
</paragraph>
<paragraph>
<text>part 2 initial mode query (priority -1) matching M</text>
</paragraph>
<paragraph>
<text>part 2 initial mode id matching N</text>
</paragraph>
<paragraph>
<text>part 2 initial mode element matching O</text>
</paragraph>
<paragraph>
<text>part 2 initial mode default matching P</text>
</paragraph>
</simple-page-sequence>
</fot>
